snuba/query/logical.py

```diff
@@ -1,15 +1,14 @@
 from __future__ import annotations
 
-from typing import Callable, Iterable, Optional, Sequence
+from abc import ABCMeta
+from typing import Any, Callable, Iterable, Optional, Sequence, Type, cast
 
-from snuba.query import LimitBy, OrderBy
-from snuba.query import ProcessableQuery as AbstractQuery
-from snuba.query import SelectedExpression
-from snuba.query.data_source.simple import Entity
+from snuba.query import LimitBy, OrderBy, ProcessableQuery, SelectedExpression
+from snuba.query.data_source.simple import Entity, LogicalDataSource, Storage
 from snuba.query.expressions import Expression, ExpressionVisitor
 
 
-class Query(AbstractQuery[Entity]):
+class Query(ProcessableQuery[LogicalDataSource]):
     """
     Represents the logical query during query processing.
```

Comment:
Could we add as a comment that a logical query can either be a logical entity query or a logical storage query for clarity?

Reply:
fosho